### ISSUE TITLE
Question regarding quality of very extensive scenes

### ISSUE THREAD

**Issue by Ph03n1xdust** [graphdeco-inria/gaussian-splatting#468]
Hi all!

Thank you for the great work and also open-sourcing it! I would like to have a question about training on very large scenes. A big advantage of this method compared to NERF could be, that you do not necessarily have to "blockify" it, as you have to do in the case of block-NERFs due to the limited capacity of their NNs.

But when I try to train on a large scenes I get significantly worse results compared to when I just pick a small portion of the scene and train on that. Of course for the large scenes (~8k images) I scale up the training iterations and the position_lr_max_steps, opacity resets, densification intervals, etc... accordingly approximately linearly with the number of images (assuming that the original datasets contained ~300 images). I also guessed that I have to scale down the densification threshold (since there would be training steps from very far from the gaussians --> smaller grads --> lower average grads, and also because the final ply contains only ~1-2 million gaussians with the default, and I would expect way more for the large scenes)

Did anyone experiment with something like this, and if yes did you run into the same issue / did you manage to find a trick to solve it?

Thanks for the help in advance!

**Comment by jaco001** [comment]
Personally, I prefer to build the stage from small pieces. You can train them separately, correct them and turn them on when necessary. A large stage takes a lot of time in training and improving it takes a lot of time too. 
We will wait for new tools and maybe the difference will not be that big, but this is the future.

**Comment by dxv2k** [comment]
Do you guys have any suggestions on where to start with the idea of splitting the scene into smaller sections?

**Comment by igarzonpres** [comment]
> Hi all!
> 
> Thank you for the great work and also open-sourcing it! I would like to have a question about training on very large scenes. A big advantage of this method compared to NERF could be, that you do not necessarily have to "blockify" it, as you have to do in the case of block-NERFs due to the limited capacity of their NNs.
> 
> But when I try to train on a large scenes I get significantly worse results compared to when I just pick a small portion of the scene and train on that. Of course for the large scenes (~8k images) I scale up the training iterations and the position_lr_max_steps, opacity resets, densification intervals, etc... accordingly approximately linearly with the number of images (assuming that the original datasets contained ~300 images). I also guessed that I have to scale down the densification threshold (since there would be training steps from very far from the gaussians --> smaller grads --> lower average grads, and also because the final ply contains only ~1-2 million gaussians with the default, and I would expect way more for the large scenes)
> 
> Did anyone experiment with something like this, and if yes did you run into the same issue / did you manage to find a trick to solve it?
> 
> Thanks for the help in advance!

Hi!, Did you find something interesting in this realm?

**Comment by Ph03n1xdust** [comment]
> > Hi all!
> > Thank you for the great work and also open-sourcing it! I would like to have a question about training on very large scenes. A big advantage of this method compared to NERF could be, that you do not necessarily have to "blockify" it, as you have to do in the case of block-NERFs due to the limited capacity of their NNs.
> > But when I try to train on a large scenes I get significantly worse results compared to when I just pick a small portion of the scene and train on that. Of course for the large scenes (~8k images) I scale up the training iterations and the position_lr_max_steps, opacity resets, densification intervals, etc... accordingly approximately linearly with the number of images (assuming that the original datasets contained ~300 images). I also guessed that I have to scale down the densification threshold (since there would be training steps from very far from the gaussians --> smaller grads --> lower average grads, and also because the final ply contains only ~1-2 million gaussians with the default, and I would expect way more for the large scenes)
> > Did anyone experiment with something like this, and if yes did you run into the same issue / did you manage to find a trick to solve it?
> > Thanks for the help in advance!
> 
> Hi!, Did you find something interesting in this realm?

Hey!

I am still working with blocks of the scene with significant overlap between them, this seems to work well, but there are quite some papers regarding large scale GS, like: 
- https://arxiv.org/abs/2505.23158
- https://github.com/city-super/Scaffold-GS
- probably many others, these were just from the top of my head

If you happen to find something good or test these it I would be interested in your opinion about which is better for you.

**Comment by igarzonpres** [comment]
> > > Hi all!
> > > Thank you for the great work and also open-sourcing it! I would like to have a question about training on very large scenes. A big advantage of this method compared to NERF could be, that you do not necessarily have to "blockify" it, as you have to do in the case of block-NERFs due to the limited capacity of their NNs.
> > > But when I try to train on a large scenes I get significantly worse results compared to when I just pick a small portion of the scene and train on that. Of course for the large scenes (~8k images) I scale up the training iterations and the position_lr_max_steps, opacity resets, densification intervals, etc... accordingly approximately linearly with the number of images (assuming that the original datasets contained ~300 images). I also guessed that I have to scale down the densification threshold (since there would be training steps from very far from the gaussians --> smaller grads --> lower average grads, and also because the final ply contains only ~1-2 million gaussians with the default, and I would expect way more for the large scenes)
> > > Did anyone experiment with something like this, and if yes did you run into the same issue / did you manage to find a trick to solve it?
> > > Thanks for the help in advance!
> > 
> > 
> > Hi!, Did you find something interesting in this realm?
> 
> Hey!
> 
> I am still working with blocks of the scene with significant overlap between them, this seems to work well, but there are quite some papers regarding large scale GS, like:
> 
> * https://arxiv.org/abs/2505.23158
> * https://github.com/city-super/Scaffold-GS
> * probably many others, these were just from the top of my head
> 
> If you happen to find something good or test these it I would be interested in your opinion about which is better for you.

Hey! Thanks for the references. 

I was having some trouble selecting which hyperparameters and how to modify when having large scenes (I have around 3.6k images and 6M initial sparse point cloud (as generated by COLMAP). I was wondering if you coul provide some advise on that. 

Moreover, why did you choose to render chunks of the scenes and not to optimize the training for the large scene at once? It is optimization issue or hardware limitation?

I'll provide the insights with the code of the papers you provided when I try them. 

Thanks!
